lib/forwarder.js

```diff
@@ -38,9 +71,24 @@ export async function forwardToDaemon(resolver, config) {
     );
   }
 
-  const socket = net.connect(config.port, '127.0.0.1');
+  let socket;
+  try {
+    socket = await connectToDaemon(config);
+  } catch (/** @type {any} */ err) {
+    if (err['code'] === 'ECONNREFUSED') {
```

Comment:
in 99% cases this is the only reason promise rejects, also here we didn't sent any data to socket yet, already read stdin and didn't write anything to connection, I think it's good place to integrate restart logic